tests/testthat/test-read_vpts.R

```diff
@@ -1,3 +1,180 @@
+      lon = -180
+    ),
+    regexp = "'radar' argument missing. Required to specify a radar identifier.",
+    fixed = TRUE
+  )
+  expect_error(
+    read_vpts(vptsfile,
+      lon = -1080,
+      lat = 38
+    ),
+    regexp = "lon' should be numeric between -360 and 360 degrees",
+    fixed = TRUE
+  )
+  expect_error(
+    read_vpts(vptsfile,
+      height = -1
+    ),
+    regexp = "'height' should be a positive number of meters above sea level",
+    fixed = TRUE
+  )
+  expect_error(
+    read_vpts(vptsfile,
+      height = NA
+    ),
+    regexp = "height is not a number (a length one numeric vector).",
+    fixed = TRUE
+  )
+  expect_error(
+    read_vpts(vptsfile,
+      height = NA
+    ),
+    regexp = "height is not a number (a length one numeric vector).",
+    fixed = TRUE
+  )
+  expect_error(
+    read_vpts(vptsfile,
+      radar = "KBGM",
+      height = seq(50)
+    ),
+    regexp = "height is not a number (a length one numeric vector).",
+    fixed = TRUE
+  )
+
+  wavelength_msg <-
+    glue::glue(
+      "'wavelength' should be a single positive number",
+      ", or one of 'C' or 'S' for C-band and S-band radar, respectively."
+    )
+  expect_error(
+    read_vpts(vptsfile,
+      wavelength = "a",
+      radar = "KBGM"
+    ),
+    regexp = wavelength_msg,
+    fixed = TRUE
+  )
+  expect_error(
+    read_vpts(vptsfile,
+      wavelength = -12,
+      radar = "KBGM"
+    ),
+    regexp = wavelength_msg,
+    fixed = TRUE
+  )
+  expect_error(
+    read_vpts(vptsfile,
+      wavelength = 1:3,
+      radar = "KBGM"
+    ),
+    regexp = wavelength_msg,
+    fixed = TRUE
+  )
+  expect_error(
+    read_vpts(vptsfile,
+      wavelength = "Q",
+      radar = "KBGM"
+    ),
+    regexp = wavelength_msg,
+    fixed = TRUE
+  )
+  expect_error(
+    read_vpts(vptsfile,
+      wavelength = "S",
+      radar = "KBGM",
+      sep = "|"
+    ),
+    regexp = "'sep' should be either \",\" or \"\"",
+    fixed = TRUE
+  )
+  expect_error(
+    read_vpts(vptsfile,
+      sep = NA
+    ),
+    regexp = "'sep' should be either \",\" or \"\"",
+    fixed = TRUE
+  )
+  expect_error(
+    read_vpts(vptsfile,
+      sep = c(",", "&")
+    ),
+    regexp = "'sep' should be either \",\" or \"\"",
+    fixed = TRUE
+  )
+  expect_error(
+    read_vpts(vptsfile,
+      sep = c(",", "||")
+    ),
+    regexp = "'sep' should be either \",\" or \"\"",
+    fixed = TRUE
+  )
+})
 
+test_that("read_vpts() warns for missing wavelength", {
+  expect_warning(
+    read_vpts(vptsfile,
+      radar = "KBGM"
+    ),
+    regexp = "No 'wavelength' argument provided, assuming radar operates at C-band",
+    fixed = TRUE
+  )
+  # test a vptsfile with a comma seperator
+  vptsfile_lines <- readLines(vptsfile)
+  vptsfile_comment_lines <- grepl("#", readLines(vptsfile))
+  vptsfile_lines[!vptsfile_comment_lines] <-
+    gsub("\\s+", ",", readLines(vptsfile)[!vptsfile_comment_lines])
+  vptsfile_sep <- tempfile()
+  writeLines(vptsfile_lines, vptsfile_sep)
```

Comment:
incomplete?

Reply:
Well spotted, a train of thought I didn't complete. Testing a file with a different separator is something to consider for the future.